examples/light-switch-app/nrfconnect/main/AppTask.cpp

```diff
@@ -0,0 +1,601 @@
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+#include <app/clusters/ota-requestor/BDXDownloader.h>
+#include <app/clusters/ota-requestor/DefaultOTARequestorStorage.h>
+#include <app/clusters/ota-requestor/GenericOTARequestorDriver.h>
+#include <app/clusters/ota-requestor/OTARequestor.h>
+#include <platform/nrfconnect/OTAImageProcessorImpl.h>
+#endif
+
+#include <dk_buttons_and_leds.h>
+#include <logging/log.h>
+#include <system/SystemError.h>
+#include <zephyr.h>
+
+using namespace ::chip;
+using namespace ::chip::app;
+using namespace ::chip::Credentials;
+using namespace ::chip::DeviceLayer;
+
+LOG_MODULE_DECLARE(app);
+namespace {
+constexpr EndpointId kLightSwitchEndpointId           = 1;
+static constexpr uint32_t kFactoryResetTriggerTimeout = 3000;
+static constexpr uint32_t kFactoryResetCancelWindow   = 3000;
+static constexpr uint32_t kDimmerTriggeredTimeout     = 500;
+static constexpr uint32_t kDimmerInterval             = 300;
+static constexpr size_t kAppEventQueueSize            = 10;
+
+K_MSGQ_DEFINE(sAppEventQueue, sizeof(AppEvent), kAppEventQueueSize, alignof(AppEvent));
+
+LightSwitch sLightSwitch;
+
+LEDWidget sStatusLED;
+LEDWidget sDiscoveryLED;
+LEDWidget sBleLED;
+LEDWidget sUnsedLED;
+
+static bool sIsThreadProvisioned    = false;
+static bool sIsThreadEnabled        = false;
+static bool sIsThreadBLEAdvertising = false;
+static bool sIsSMPAdvertising       = false;
+static bool sHaveBLEConnections     = false;
+static bool sIsDiscoveryEnabled     = false;
+static bool sWasDimmerTriggered     = false;
+
+k_timer sFunctionTimer;
+k_timer sDimmerPressKeyTimer;
+k_timer sDimmerTimer;
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+DefaultOTARequestorStorage sRequestorStorage;
+GenericOTARequestorDriver sOTARequestorDriver;
+OTAImageProcessorImpl sOTAImageProcessor;
+chip::BDXDownloader sBDXDownloader;
+chip::OTARequestor sOTARequestor;
+#endif
+
+} /* namespace */
+
+AppTask AppTask::sAppTask;
+
+CHIP_ERROR AppTask::Init()
+{
+    /* Initialize CHIP */
+    LOG_INF("Init CHIP stack");
+
+    CHIP_ERROR err = chip::Platform::MemoryInit();
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("Platform::MemoryInit() failed");
+        return err;
+    }
+
+    err = PlatformMgr().InitChipStack();
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("PlatformMgr().InitChipStack() failed");
+        return err;
+    }
+
+    err = ThreadStackMgr().InitThreadStack();
+    if (err != CHIP_NO_ERROR)
+    {
+        LOG_ERR("ThreadStackMgr().InitThreadStack() failed: %s", chip::ErrorStr(err));
+        return err;
+    }
+
+#ifdef CONFIG_OPENTHREAD_MTD
+#ifdef CONFIG_OPENTHREAD_MTD_SED
+    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_SleepyEndDevice);
+#else
+    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
+#endif
+#else
+    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_FullEndDevice);
+#endif
```

Comment:
How about this one to not nest ifdefs?
```suggestion
#ifdef CONFIG_OPENTHREAD_MTD_SED
    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_SleepyEndDevice);
#elif CONFIG_OPENTHREAD_MTD
    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_MinimalEndDevice);
#else
    err = ConnectivityMgr().SetThreadDeviceType(ConnectivityManager::kThreadDeviceType_FullEndDevice);
#endif
```